src/linker/Linker.Dataflow/MethodBodyScanner.cs

```diff
@@ -187,7 +187,8 @@ static void ValidateNoReferenceToReference (LocalVariableStore locals, MethodDef
 				VariableDefinition localVariable = keyValuePair.Key;
 				foreach (var val in localValue) {
 					if (val is LocalVariableReferenceValue reference
-						&& locals[reference.LocalDefinition].Value.Any (v => v is ReferenceValue)) {
+					&& locals.TryGetValue (reference.LocalDefinition, out var referencedValue)
+					&& referencedValue.Value.Any (v => v is ReferenceValue)) {
```

Comment:
So basically this new logic says continue, in the scenario that was failing:
V_9 was trying to reference V_2 which was not present in the locals at that moment. What happens if we continue and then V_2 was a ReferenceValue, we are long gone from the check of V_9, is that even possible?